File: backend/src/main/java/com/grupo1/backend/entities/Pedido.java
package com.grupo1.backend.entities;

import java.sql.Date;
import java.util.List;

import org.hibernate.annotations.ManyToAny;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.grupo1.backend.entities.enums.MetodoPago;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "pedidos")
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "fecha")
    private Date fecha;

    @Column(name = "metodo_pago")
    @Enumerated(EnumType.STRING)
    private MetodoPago metodo_pago;

    @Column(name = "direccion")
    private String direccon;

    @ManyToOne
    @JoinColumn(name = "user")
    @JsonIgnore
    private User user;

    @ManyToAny
    @JoinTable(
        name = "pedido_producto",
        joinColumns = @JoinColumn(name = "id_pedido"),
        inverseJoinColumns = @JoinColumn(name = "id_producto")
    )
    private List<Producto> productos;
}

File: frontEnd/src/app/models/pedido.ts
import { MetodoPago } from "./enums/MetodoPago.enum";
import { Producto } from "./producto";
import { User } from "./user";

export class Pedido {
    id?: number;
    fecha: string; 
    metodo_pago: MetodoPago; 
    direccon: string;
    productos: Producto[];
    user?: User;

    constructor(data?: Partial<Pedido>){
        this.id = data?.id;
        this.fecha = data?.fecha ?? new Date().toISOString().split('T')[0];
        this.metodo_pago = data?.metodo_pago ?? MetodoPago.PAYPAL;
        this.direccon = data?.direccon ?? '';
        this.productos = data?.productos ?? [];
        this.user = data?.user; 
    }
}